tests/admin_views/tests.py

```diff
@@ -1508,6 +1508,15 @@ def test_change_view_subtitle_per_object(self):
         self.assertContains(response, "<h1>Change article</h1>")
         self.assertContains(response, "<h2>Article 2</h2>")
 
+    def test_change_view_error_subtitle_with_object(self):
```

Comment:
To check a couple of templates and title formats, maybe let's add a subtest for the add page and login page

Reply:
So the prefix of "Error:" in the titlte should be there for Login page also? As of now its there only for Add and Update pages

Reply:
So I think, in terms of helping screen reader users, it would make sense to do this
Does mean maybe we should update the title in specific cases rather than the base template
I made these updates as an example:
```diff
diff --git a/django/contrib/admin/sites.py b/django/contrib/admin/sites.py
index 88d53f5f4c..dc67262afc 100644
--- a/django/contrib/admin/sites.py
+++ b/django/contrib/admin/sites.py
@@ -339,7 +339,6 @@ class AdminSite:
             "is_popup": False,
             "is_nav_sidebar_enabled": self.enable_nav_sidebar,
             "log_entries": self.get_log_entries(request),
-            "errors": False,
         }
 
     def password_change(self, request, extra_context=None):
diff --git a/django/contrib/admin/templates/admin/base_site.html b/django/contrib/admin/templates/admin/base_site.html
index 49d1518288..4405998164 100644
--- a/django/contrib/admin/templates/admin/base_site.html
+++ b/django/contrib/admin/templates/admin/base_site.html
@@ -1,6 +1,6 @@
 {% extends "admin/base.html" %}
 
-{% block title %}{% if errors %}{{ _('Error:') }} {% endif %}{% if subtitle %}{{ subtitle }} | {% endif %}{{ title }} | {{ site_title|default:_('Django site admin') }}{% endblock %}
+{% block title %}{% if subtitle %}{{ subtitle }} | {% endif %}{{ title }} | {{ site_title|default:_('Django site admin') }}{% endblock %}
 
 {% block branding %}
 <div id="site-name"><a href="{% url 'admin:index' %}">{{ site_header|default:_('Django administration') }}</a></div>
diff --git a/django/contrib/admin/templates/admin/change_form.html b/django/contrib/admin/templates/admin/change_form.html
index 31ff5d6c10..0f153cdc26 100644
--- a/django/contrib/admin/templates/admin/change_form.html
+++ b/django/contrib/admin/templates/admin/change_form.html
@@ -1,6 +1,8 @@
 {% extends "admin/base_site.html" %}
 {% load i18n admin_urls static admin_modify %}
 
+{% block title %}{% if errors %}{{ _('Error:') }} {% endif %}{{ block.super }}{% endblock %}
+
 {% block extrahead %}{{ block.super }}
 <script src="{% url 'admin:jsi18n' %}"></script>
 {{ media }}
diff --git a/django/contrib/admin/templates/admin/login.html b/django/contrib/admin/templates/admin/login.html
index b61d9ec603..8389db31c2 100644
--- a/django/contrib/admin/templates/admin/login.html
+++ b/django/contrib/admin/templates/admin/login.html
@@ -1,6 +1,8 @@
 {% extends "admin/base_site.html" %}
 {% load i18n static %}
 
+{% block title %}{% if form.errors %}{{ _('Error:') }} {% endif %}{{ block.super }}{% endblock %}
+
 {% block extrastyle %}{{ block.super }}<link rel="stylesheet" href="{% static "admin/css/login.css" %}">
 {{ form.media }}
 {% endblock %}
diff --git a/tests/admin_views/tests.py b/tests/admin_views/tests.py
index 22e65933ff..4fa685402e 100644
--- a/tests/admin_views/tests.py
+++ b/tests/admin_views/tests.py
@@ -1488,6 +1488,28 @@ class AdminViewBasicTest(AdminViewBasicTestCase):
         models = [model["name"] for model in response.context["app_list"][0]["models"]]
         self.assertSequenceEqual(models, sorted(models, reverse=True))
 
+    def test_error_subtitle(self):
+        for url, title, client in [
+            (
+                reverse("admin:admin_views_article_change", args=(self.a1.pk,)),
+                "Error: Article 1 | Change article | Django site admin",
+                self.client,
+            ),
+            (
+                reverse("admin:admin_views_article_add"),
+                "Error: Add article | Django site admin",
+                self.client,
+            ),
+            (
+                reverse("admin:login"),
+                "Error: Log in | Django site admin",
+                self.client_class(),
+            ),
+        ]:
+            with self.subTest(url=url, title=title):
+                response = self.client.post(url, {})
+                self.assertContains(response, title)
+
     def test_change_view_subtitle_per_object(self):
         response = self.client.get(
             reverse("admin:admin_views_article_change", args=(self.a1.pk,)),
@@ -1508,15 +1530,6 @@ class AdminViewBasicTest(AdminViewBasicTestCase):
         self.assertContains(response, "<h1>Change article</h1>")
         self.assertContains(response, "<h2>Article 2</h2>")
 
-    def test_change_view_error_subtitle_with_object(self):
-        response = self.client.post(
-            reverse("admin:admin_views_article_change", args=(self.a1.pk,)), {}
-        )
-        self.assertContains(
-            response,
-            "<title>Error: Article 1 | Change article | Django site admin</title>",
-        )
-
     def test_view_subtitle_per_object(self):
         viewuser = User.objects.create_user(
             username="viewuser",
```
Are there other templates which involve a form which need an error prefix do you think?

Reply:
Is it better to add the "Error:" prefix in the title separatly in each templates? Then we need to add it in the `django/contrib/admin/templates/admin/change_list.html` template also i guess along with `change_form.html` and `login.html`

Reply:
I think one other way to do this is by, having "errors" context variable in every template and adding the "Error:" prefix in the `base_site.html` template.

If we want to add it separatly in the templates, then we might need to add them in the following templates I guess
- `django/contrib/admin/templates/admin/change_form.html`
- `django/contrib/admin/templates/admin/change_list.html`
- `django/contrib/admin/templates/admin/login.html`
- `django/contrib/admin/templates/admin/auth/user/change_password.html`
- `django/contrib/admin/templates/registration/password_change_form.html`
- `django/contrib/admin/templates/registration/password_reset_confirm.html`
- `django/contrib/admin/templates/registration/password_reset_form.html`

So let me know which method is better?

Reply:
Each of those needs a review. The ones where it is needed should have a test
I wouldn't expect to need it on `django/contrib/admin/templates/registration/password_reset_confirm.html` for example

If we're talking about almost all admin templates, then it might be better in the base template but you might need to add specific updates for "error" as on the login page that was "form.errors" (hence why a test covering each of the views is required). If we're talking about adding this to 4 templates, it might be better just to do that

Reply:
Okay make sense, So then I'll go with adding them in the templates `change_form.html`,  `login.html`, `change_list.html`. Is that fine

Reply:
I think this going to be roughly half of our templates so it's 50/50 with the approach
You probably want to do this to all of those templates you suggested

Reply:
Hmm.. I think we can go with adding them separatly in the templates only(Bcz it's less error-proof, I think).

Reply:
Sounds a plan 👍

Reply:
How can I write a test for change_list? When does error occurs in the change list page? (Is it when deleting some objects)?

Reply:
I think you're right. There might be an error if you didn't select a row/object but selected to "delete selected objects"

Reply:
Actually its just a warning message and not an error

Reply:
Actually I was not able to add tests for few templates. I have added for 4 templates. Kindly review and get me the feedback